### ISSUE TITLE
RequestTracking is not logging correct timestamps

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.webapi#333]
I have created a new simple Web API project using the Arcus Web Api v0.8 template.

In this API, I'm using Arcus request-tracking, which has been enabled like this:

```csharp
private static void ConfigureApp(IApplicationBuilder app)
{
    app.UseHttpCorrelation();
    app.UseRouting();
    app.UseRequestTracking(options => options.OmittedRoutes.Add("/"));
    app.UseExceptionHandling();
  ...
}
```

I've created a simple controller with an operation which logs a Sql Dependency.  Something as simple as this:

```csharp
[HttpPost]
public IActionResult CreateCar()

    using (var m = DurationMeasurement.Start())
    {
         _logger.LogSqlDependency("test", "database", "something", "foo", true, m);
    }
    return Ok();
}
```

When I look in AppInsights, the request tracking is wrong:
![image](https://user-images.githubusercontent.com/3605786/172628479-005cbc25-8ef4-4b9e-b72d-576fc9a7b8d3.png)

It looks like the request started later then the dependency that was logged, but that is not the case.

I'm using Arcus.Observability 2.4.0 and Arcus.WebApi 1.5.0

**Comment by fgheysels** [comment]
I believe this is related to this: https://github.com/arcus-azure/arcus.observability/issues/356

**Comment by fgheysels** [comment]
Before we can fix this, we need to create a new release for Arcus.Observability.

**Comment by stijnmoreels** [comment]
Closing in favor of #310
